### PR TITLE
New doc | Managing secrets

### PR DESCRIPTION
**Pull request by dbradish-microsoft**

### CODE REVIEW

File: docs-ref-conceptual/azure-cli-manage-secrets.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+# Use Azure CLI to manage sensitive Azure information such as keys, passwords and secrets
+
+When you manage Azure resources, there is sensitive information that must be protected. For example, keys, passwords and connection strings can be created by Azure CLI commands and displayed in a terminal window.
+It is critical to protect this information which, when acquired publically, can cause serious damage and lead to a loss of trust in your company's products and services.
+
+## Set show_secrets_warning (phase 1)
+
+Beginning in Azure CLI x.y.z, a warning message will display when reference commands result in the output of sensitive information. The purpose of the warning message is to decrease the unintentional exposure of secrets, but these messages may require you to make changes in existing scripts.
+
```
Comment (dcaro) [inline docs-ref-conceptual/azure-cli-manage-secrets.md:21]:
```suggestion
Beginning in Azure CLI 2.57, a warning message can be displayed when reference commands result in the output of sensitive information. The purpose of the warning message is to decrease the unintentional exposure of secrets, but these messages may require you to make changes in existing scripts.

```

File: docs-ref-conceptual/azure-cli-manage-secrets.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+> The new warning messages are sent to Standard Error (STDERR), not Standard Out (STDOUT).
+> Therefore, if you are running an Azure CLI command that results in sensitive information output, you will need to trap for the warning message or turn warnings off using `az config set show_warnings = off`.
+
+The Azure CLI identifies "sensitive information" by **{infomation from engineering here}.**
+
+Turn sensitive information warnings `on` and `off` by setting the `show_secrets_warning` configuration property.
+
+```azurecli-interactive
+az config set show_warnings = off
+```
```
Comment (dcaro) [inline docs-ref-conceptual/azure-cli-manage-secrets.md:32]:
```suggestion
```azurecli-interactive
az config set clients.show_secrets_warning=True
```
```

File: docs-ref-conceptual/azure-cli-manage-secrets.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+> [!NOTE]
+> Warnings are **enabled** by default in Azure Cloud Shell. If you run the Azure CLI locally through Cmd or PowerShell, warnings are **disabled** by default.
```
Comment (dcaro) [inline docs-ref-conceptual/azure-cli-manage-secrets.md:35]:
```suggestion
> Warnings are **enabled** by default in Azure Cloud Shell. If you run the Azure CLI locally through any supported shell (PowerShell or zsh for example), warnings are **disabled** by default.
```

**Review by dcaro — COMMENTED** [review]
Thanks for this first version, I made few suggestions.
Could we also add references to the following articles:
- https://learn.microsoft.com/en-us/cli/azure/format-output-azure-cli?tabs=bash#none-output-format
- https://msrc.microsoft.com/blog/2023/11/microsoft-guidance-regarding-credentials-leaked-to-github-actions-logs-through-azure-cli/

File: docs-ref-conceptual/azure-cli-manage-secrets.md (hunk 2 above)
Comment (dbradish-microsoft) [reply docs-ref-conceptual/azure-cli-manage-secrets.md]: @dcaro, in [azure-cli-configuration.md](https://learn.microsoft.com/en-us/cli/azure/azure-cli-configuration?view=azure-cli-latest) we use `yes` and `no`, so I changed `off` to `yes`.

**Comment by dbradish-microsoft** [comment]
@Dcaro, some portions of your WORD doc were incorporated into this PR yesterday, Feb 6.  I then took your feedback and wove it into yesterday's updates.  This PR is ready for final review.

**Comment by dbradish-microsoft** [comment]
Merging per @dcaro's request.

**Pull request merged**